modules/models/src/main/java/org/eclipse/xpanse/modules/models/utils/DeployVariableValidator.java

```diff
@@ -0,0 +1,317 @@
+                String[] keyValueArray = StringUtils.split(validStr, "=", 2);
+                if (keyValueArray.length == 2) {
+                    String key = keyValueArray[0];
+                    String value = keyValueArray[1];
+                    try {
+                        // check validator key.
+                        VariableValidator validator =
+                                VariableValidator.valueOf(StringUtils.upperCase(key));
+                        // check number validator.
+                        if (DeployVariableType.NUMBER.equals(type)) {
+                            if (validator.equals(VariableValidator.MINIMUM)
+                                    || validator.equals(VariableValidator.MAXIMUM)) {
+                                if (value.matches(NUMBER_REGEX)) {
+                                    validatorMap.put(validator, value);
+                                }
+                            } // check string validator.
+                        } else if (DeployVariableType.STRING.equals(type)) {
+                            if (validator.equals(VariableValidator.MINLENGTH)
+                                    || validator.equals(VariableValidator.MAXLENGTH)) {
+                                if (Integer.parseInt(value) >= 0) {
+                                    validatorMap.put(validator, value);
+                                }
+                            } else if (validator.equals(VariableValidator.ENUM)) {
+                                if (value.startsWith("[") && value.endsWith("]")) {
+                                    validatorMap.put(validator, value);
+                                }
+
+                            } else if (validator.equals(
+                                    VariableValidator.PATTERN)) {
+                                if (key.matches(value)) {
+                                    validatorMap.put(validator, value);
+                                }
+                            }
+                        }
+                    } catch (IllegalArgumentException e) {
+                        log.error("Variable:{} contains a invalid validaStr:{}", variableName,
+                                validStr);
+                    }
+                }
+            }
+        }
+        return validatorMap;
+    }
+
+
+    /**
+     * Check validation of deploy property map by list of deployVariables in registered service.
+     *
+     * @param deployVariables list of deployVariables in registered service
+     * @param deployProperty  deploy property map
+     * @return isValid
+     */
+    public boolean checkVariableValidation(List<DeployVariable> deployVariables,
+            Map<String, String> deployProperty) {
+
+        if (!CollectionUtils.isEmpty(deployVariables) && Objects.nonNull(deployProperty)) {
+            // check required keys
+            Set<String> userPutKeys = deployProperty.keySet();
+            log.error("userPutKeys:{}", userPutKeys);
```

Comment:
is log level error correct here? May be DEBUG is better?

Reply:
Yes.